src/api/frontend.c

```diff
@@ -304,6 +309,27 @@ EXPORT m64p_error CALL CoreDoCommand(m64p_command Command, int ParamInt, void *P
                 return M64ERR_INPUT_INVALID;
             g_media_loader = *(m64p_media_loader*)ParamPtr;
             return M64ERR_SUCCESS;
+        case M64CMD_NETPLAY_INIT:
+            return netplay_start(ParamPtr, ParamInt);
+        case M64CMD_NETPLAY_CONTROL_PLAYER:
+            response = netplay_register_player(ParamInt - 1);
+            if (response == 1)
+            {
+                netplay_set_controller(ParamInt - 1);
+                return M64ERR_SUCCESS;
+            }
+            else if (response == -1)
+                return M64ERR_SYSTEM_FAIL; // no response from server
+            else
+                return M64ERR_INPUT_INVALID; // player already in use
+        case M64CMD_NETPLAY_GET_VERSION:
+            memcpy(ParamPtr, git_version, 41);
+            if (ParamInt == NETPLAY_API_VERSION)
+                return M64ERR_SUCCESS;
+            else
+                return M64ERR_INCOMPATIBLE;
```

Comment:
I'm not sure using git_version for this is a good idea. Some distributions could have security patches or anything that could make this fail.

Reply:
I needed some way to verify that each client is using the exact same version of the core. Just checking for "version 2.5.9" won't cut it unfortunately.

This does mean that it would be almost impossible for somebody that just pulls mupen64plus from the Arch Linux repo to play with someone that got their mupen64plus from somewhere else, but that is how it should be, the clients will likely desync if they aren't running the exact same version.